subsys/net/ip/connection.c

```diff
@@ -459,6 +459,7 @@ static bool conn_addr_cmp(struct net_pkt *pkt,
 	return true;
 }
 
+#if !defined(CONFIG_NET_DISABLE_ICMP_DESTINATION_UNREACHABLE)
 static inline void conn_send_icmp_error(struct net_pkt *pkt)
 {
 	if (IS_ENABLED(CONFIG_NET_IPV6) && net_pkt_family(pkt) == AF_INET6) {
```

Comment:
wouldn't it be better to enclose the body of this function in:

```
if (!IS_ENABLED(CONFIG_NET_DISABLE_ICMP_DESTINATION_UNREACHABLE)) {
```

Reply:
Sure, absolutely - the `#if !defined(...)` style was [mentioned](https://github.com/zephyrproject-rtos/zephyr/pull/37345#discussion_r680788595) in an earlier review comment, but I see that `IS_ENABLED()` is more pervasive in the networking stack.

Reply:
@rlubos do you have an additional opinion on this ☝️  regarding dummy "stub" vs using `IS_ENABLED()` within the `conn_send_icmp_error()` body?

Reply:
@berendo - I would agree with @carlescufi in this case. The main advantage of using `IS_ENABLED()` (aside from aesthetics) is that those lines are still compiled, which prevents bugs from sneaking in sometimes.

Reply:
Well, both constructs are used widely across the networking stack, it seems that in this particular case it's ok to use runtime check as well. I'd rather however exit early at the beginning of the function instead of adding an extra indentation to the whole body, i. e:
```
if (IS_ENABLED(CONFIG_NET_DISABLE_ICMP_DESTINATION_UNREACHABLE)) {
	return;
}
```